core/src/main/java/nu/yona/server/analysis/service/ActivityService.java

```diff
@@ -145,17 +149,22 @@ private LocalDate getCurrentDayDate(UserAnonymizedDTO userAnonymized)
 
 	private <T extends IntervalActivity> void addMissingInactivity(Goal activeGoal, ZonedDateTime dateAtStartOfInterval,
 			Set<T> activityEntitiesAtDate, UserAnonymizedDTO userAnonymized,
-			BiFunction<Goal, ZonedDateTime, T> inactivityEntitySupplier)
+			BiFunction<Goal, ZonedDateTime, T> inactivityEntitySupplier, Consumer<T> existingEntityInactivityCompletor)
 	{
-		if (!containsActivityForGoal(activityEntitiesAtDate, activeGoal))
+		Optional<T> activityForGoal = getActivityForGoal(activityEntitiesAtDate, activeGoal);
+		if (activityForGoal.isPresent())
+		{
+			existingEntityInactivityCompletor.accept(activityForGoal.get());
```

Comment:
I don't understand what's going on here. The method is named "add missing inactivity", but you're here adding "activity" to an "inactivity". Can you explain?